code/game/machinery/syndicatebeacon.dm

```diff
@@ -102,119 +102,163 @@
-				anchored = 0
-				to_chat(user, "<span class='notice'>You unscrew the beacon from the floor.</span>")
-				attached = null
-				return
-			else
-				var/turf/T = loc
-				if(isturf(T) && !T.intact)
-					attached = locate() in T
-				if(!attached)
-					to_chat(user, "This device must be placed over an exposed cable.")
-					return
-				stat |= SCREWED
-				anchored = 1
-				to_chat(user, "<span class='notice'>You screw the beacon to the floor and attach the cable.</span>")
-				return
-		..()
+	if(active)
+		to_chat(user, "<span class='warning'>Turn off \the [src] first.</span>")
 		return
+	..()
+	if(attached)
+		attached = null //Reset attached cable
+
+/obj/machinery/singularity_beacon/Destroy()
+	if(active)
+		deactivate()
+	..()
+
+/*
+* Added for a simple way to check power. Verifies that the beacon
+* is connected to a wire, the wire is part of a powernet (that part's
+* sort of redundant, since all wires either join or create one when placed)
+* and that the powernet has at least 1500 power units available for use.
+* Doesn't use them, though, just makes sure they're there.
+* - QualityVan, Aug 11 2012
+*/
+
+//Simplified check for power. If we can charge straight out of the grid, do it
+/obj/machinery/singularity_beacon/proc/check_wire_power()
+	if(!attached) //No wire, move straight to battery power
+		return 0
+	var/datum/powernet/PN = attached.get_powernet()
+	if(!PN) //Powernet is dead
+		return 0
+	if(PN.avail < power_load) //Cannot drain enough power, needs 1500 per tick, move to battery
+		return 0
+	else
+		PN.load += power_load
+		if(cell && cell.charge < cell.maxcharge && cell.charge > 0 && PN.netexcess)
+			power_draw = min(cell.maxcharge - cell.charge, PN.netexcess) //Draw power directly from excess power
+			PN.load += power_draw
+			cell.give(power_draw) //We drew power from the grid, charge the cell
+		return 1
 
+//Use up the battery if powernet check fails
+/obj/machinery/singularity_beacon/proc/check_battery_power()
 
-	Destroy()
-		if(active) Deactivate()
-		..()
-
-	/*
-	* Added for a simple way to check power. Verifies that the beacon
-	* is connected to a wire, the wire is part of a powernet (that part's
-	* sort of redundant, since all wires either join or create one when placed)
-	* and that the powernet has at least 1500 power units available for use.
-	* Doesn't use them, though, just makes sure they're there.
-	* - QualityVan, Aug 11 2012
-	*/
-	proc/checkWirePower()
-		if(!attached)
-			return 0
-		var/datum/powernet/PN = attached.get_powernet()
-		if(!PN)
-			return 0
-		if(PN.avail < 1500)
-			return 0
+	if(cell && cell.charge > power_load)
+		cell.use(power_load)
 		return 1
+	else //Nothing here either
+		return 0
 
-	process()
-		if(!active)
-			return
-		else
-			if(!checkWirePower())
-				Deactivate()
-		return
+//Composite of the two, called at every process
+/obj/machinery/singularity_beacon/proc/check_power()
 
+	if(!check_wire_power()) //First, check the connected wire if any

```

Comment:
???

You do realise that if the first part of a `||` is true, it doesn't bother to try the second one, right?